store/src/migrations.rs

```diff
@@ -5,24 +5,59 @@ pub static MIGRATIONS: Lazy<Migrations> = Lazy::new(|| {
     Migrations::new(vec![M::up(
         "
         CREATE TABLE
-            block_header
+            block_headers
         (
             block_num INTEGER NOT NULL,
             block_header BLOB NOT NULL,
 
             PRIMARY KEY (block_num),
-            CONSTRAINT block_header_block_num_positive CHECK (block_num >= 0)
+            CONSTRAINT block_header_block_num_is_u32 CHECK (block_num >= 0 AND block_num < 4294967296)
+        ) STRICT, WITHOUT ROWID;
+
+        CREATE TABLE
+            notes
+        (
+            block_num INTEGER NOT NULL,
+            note_index INTEGER NOT NULL,
+            note_hash BLOB NOT NULL,
+            sender INTEGER NOT NULL,
+            tag INTEGER NOT NULL,
+            num_assets INTEGER NOT NULL,
+            merkle_path BLOB NOT NULL,
+
+            PRIMARY KEY (block_num, note_index),
+            CONSTRAINT notes_block_number_is_u32 CHECK (block_num >= 0 AND block_num < 4294967296),
+            CONSTRAINT notes_note_index_is_u32 CHECK (note_index >= 0 AND note_index < 4294967296),
+            CONSTRAINT notes_sender_is_felt CHECK (sender >= 0 AND sender <= 18446744069414584321),
+            CONSTRAINT notes_tag_is_felt CHECK (tag >= 0 AND tag <= 18446744069414584321),
+            CONSTRAINT notes_num_assets_is_u8 CHECK (tag >= 0 AND tag < 256),
+            FOREIGN KEY (block_num) REFERENCES block_header (block_num)
+        ) STRICT, WITHOUT ROWID;
+
+        CREATE TABLE
+            accounts
+        (
+            account_id INTEGER NOT NULL,
+            account_hash BLOB NOT NULL,
+            block_num INTEGER NOT NULL,
+
+            PRIMARY KEY (account_id),
+            CONSTRAINT accounts_block_account_id_is_felt CHECK (account_id >= 0 AND account_id < 18446744069414584321),
+            CONSTRAINT accounts_block_num_is_u32 CHECK (block_num >= 0 AND block_num < 4294967296),
+            FOREIGN KEY (block_num) REFERENCES block_header (block_num)
         ) STRICT, WITHOUT ROWID;
 
         CREATE TABLE
             nullifiers
         (
             nullifier BLOB NOT NULL,
+            nullifier_prefix INTEGER NOT NULL,
             block_number INTEGER NOT NULL,
 
             PRIMARY KEY (nullifier),
-            CONSTRAINT nullifiers_nullifier_valid_digest CHECK (length(nullifier) = 32),
-            CONSTRAINT nullifiers_block_number_positive CHECK (block_number >= 0),
+            CONSTRAINT nullifiers_nullifier_is_digest CHECK (length(nullifier) = 32),
```

Comment:
To clarify: does this mean that some `BLOB` fields in `accounts`, `notes`, and `block_headers` tables use protobuf serialization format?

Looking through these tables, it wasn't immediately clear to me which fields would these be (maybe `block_headers.block_header` and `notes.merkle_path`?)

Reply:
Basically everything except the nullifier. Maybe I should change that too, it is only using our binary format to create the nullifier tree, but that is done once during initialization, and the other uses actually need the protobuf format

Reply:
> Basically everything except the nullifier.

Does this include non-blob fields too?

I would think integers would be recorded as integers - but maybe that's not the case?

Regarding blob fields: things like nullifiers, hashes etc. are just 32 bytes which cannot be compressed - so, the format should be the same. But maybe I'm missing something?

Reply:
> I would think integers would be recorded as integers - but maybe that's not the case?

Integers should be fine

> Regarding blob fields: things like nullifiers, hashes etc. are just 32 bytes which cannot be compressed - so, the format should be the same. But maybe I'm missing something?

The encoding format is documented [here](https://protobuf.dev/programming-guides/encoding/). There are some additional metadata in the protobuf's wire format that we don't use.

Reply:
Let's create an issue to discuss this. On the one hand, it would be nice to reduce the number of times we encode/decode things. But on the other, storing protobuf formats in the DB doesn't feel right. Maybe there are options which can balance these somehow.